backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/interim/invoice/process/InterimContract_GenerateInterimInvoice.java

```diff
@@ -137,4 +114,20 @@ private InvoicingParams getInvoicingParams()
 		return InvoicingParams.ofParams(getParameterAsIParams());
 	}
 
+	private List<I_C_Flatrate_Term> streamFlatrateTermIds(@NonNull final IQueryFilter<I_C_Flatrate_Term> filter)
```

Comment:
consider renaming it to count... and returing only the count because that's what u need in `checkPreconditionsApplicable`.

Also, keep in mind that `checkPreconditionsApplicable` methods shall be as fast as possible!!!